pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
```

Comment:
> I think I actually want it on the details, so its obvious which part of the information we've validated.

That's why there's a `ValidationResult` providing the result. `ValidatedOn` holds just a timestamp of the last validation, which could fail. Maybe the name isn't cogent enough.